microsoft-edge/webview2/concepts/process-related-events.md

```diff
@@ -0,0 +1,190 @@
+
+  - **Failure kind:** `RenderProcessUnresponsive` ([COM](<link>), [.NET](<link>), [WinRT](<link>))
+  - **Failure reason:** `Unresponsive`
+
+The event will continue to be raised every set period of time as long as the process remains unresponsive. Renderer process unresponsiveness can happen for the following reasons:
+
+  * There is a **long-running script** being executed. For example, the web content in your WebView2 control might be performing a synchronous XHR, or have entered an infinite loop. Reloading ([COM](<link>), [.NET](<link>), [WinRT](<link>)) the WebView2 control might make it responsive again.
+  * The **system is busy**.
+
+As this event will be raised repeatedly, you must decide the threshold for your application to act upon it.
+
+### Gathering Details
+Your application can use and collect information from [Process Failure Details](<link>) to identify the most frequent issues for your application.
+
+> [!NOTE]
+> Some process failures might raise the `ProcessFailed` event across different WebView2 controls in your application (see [Unexpected Exits](<link>) for more information). You must decide how often to gather details and how to handle duplicates for these cases.
+
+Additionally, most process crashes will generate dumps in the [User Data Folder](user-data-folder.md), under the directory returned by `FailureReportFolderPath` ([COM](<link>), [.NET](<link>), [WinRT](<link>)). You can use these dumps to understand crashes and provide additional information when contacting the WebView2 team.
+
+## Main Browser Process Crashes
+When the main browser process in the _WebView2 Process Group_ exits unexpectedly, **both** events discussed in this document will be raised:
+
+  * `CoreWebView2.ProcessFailed`
+  * `CoreWebView2Environment.BrowserProcessExited`
+
+`ProcessFailed` will be raised for every WebView2 control in the application that has been impacted by the failure, while `BrowserProcessExited` will be raised from each CoreWebView2Environment that is associated with the failing process. The order of these events is not guaranteed.
+
+You can use the `ProcessFailed` event to handle recovery for your application. For example, by re-creating your WebView2 controls when the event is received with **failure kind** of `BrowserProcessExited`.
+
+While a `BrowserProcessexited` **event** will be raised too, this event is intended for operations involving the WebView2 Runtime resources (such as the _User Data Folder_) and lifetime (such as updates, re-launch). The _exit kind_ event argument ([COM](<link>), [.NET](<link>), [WinRT](<link>)) is provided so your application can identify this scenario and coordinate with your `ProcessFailed` event handlers. For example, you might want to prevent racing conditions that could arise from attempting recovery while also trying to remove the _User Data Folder_.
```

Comment:
```suggestion
Although a `BrowserProcessExited` **event** will be raised too, this event is intended for operations involving the WebView2 Runtime resources (such as the _User Data Folder_) and lifetime (such as updates or re-launch). The `BrowserProcessExitKind` event argument is provided so your application can identify this scenario and coordinate with your `ProcessFailed` event handlers. For example, you might want to prevent race conditions that could arise from attempting recovery while also trying to remove the user data folder.

##### [.NET/C#](#tab/dotnetcsharp)

* [CoreWebView2Environment.BrowserProcessExited Event](/dotnet/api/microsoft.web.webview2.core.corewebview2environment.browserprocessexited)
* [CoreWebView2BrowserProcessExitedEventArgs.BrowserProcessExitKind Property](/dotnet/api/microsoft.web.webview2.core.corewebview2browserprocessexitedeventargs.browserprocessexitkind)

##### [WinRT/C#](#tab/winrtcsharp)

* [CoreWebView2Environment.BrowserProcessExited Event](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2environment#browserprocessexited)
* [CoreWebView2BrowserProcessExitedEventArgs.BrowserProcessExitKind Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2browserprocessexitedeventargs#browserprocessexitkind)

##### [Win32/C++](#tab/win32cpp)

* [ICoreWebView2Environment5::add_BrowserProcessExited event](/microsoft-edge/webview2/reference/win32/icorewebview2environment5#add_browserprocessexited)
* [ICoreWebView2Environment5::remove_BrowserProcessExited event](/microsoft-edge/webview2/reference/win32/icorewebview2environment5#remove_browserprocessexited)
* [ICoreWebView2BrowserProcessExitedEventArgs::get_BrowserProcessExitKind property](/microsoft-edge/webview2/reference/win32/icorewebview2browserprocessexitedeventargs#get_browserprocessexitkind)

---


<!-- ====================================================================== -->
## See also

* [Overview of WebView2 features and APIs](./overview-features-apis.md)
* [WebView2 API Reference](../webview2-api-reference.md)

<!-- tab-set template:

##### [.NET/C#](#tab/dotnetcsharp)


##### [WinRT/C#](#tab/winrtcsharp)


##### [Win32/C++](#tab/win32cpp)


---

-->
```
* "Although" - https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/w/while
* A tab-set for API Ref links.
* A See Also section, always good.
* A tab-set template at bottom.